runtime/src/main.rs

```diff
@@ -90,33 +90,21 @@ extern "C" fn exception_handler(trap_record: &TrapRecord) {
     handle_fatal_error(caliptra_drivers::CaliptraError::RUNTIME_GLOBAL_EXCEPTION.into());
 }
 
-const NON_DCCM_NMI: u32 = 0xF000_1002;
-
 #[no_mangle]
 #[inline(never)]
 #[allow(clippy::empty_loop)]
 extern "C" fn nmi_handler(trap_record: &TrapRecord) {
-    let mut soc_ifc = unsafe { SocIfcReg::new() };
-
-    let soc_ifc_regs = soc_ifc.regs_mut();
-    let intr_block = soc_ifc_regs.intr_block_rf();
-
-    // TODO implement MMIO reg for emulator
-    if cfg!(feature = "fpga_realtime") {
-        let notif_interrupt_status = intr_block.notif_internal_intr_r();
-
-        // On wake from mailbox command, clear the interrupt status and mret to continue execution
-        if notif_interrupt_status.read().notif_cmd_avail_sts() && trap_record.mcause == NON_DCCM_NMI
-        // Machine Fast Interrupt non-DCCM region NMI
-        {
-            notif_interrupt_status.modify(|w| w.notif_cmd_avail_sts(true));
```

Comment:
Clearing the interrupt status is now never done? Does the CPU not come out of halted state even without new mailbox commands?

Reply:
I didn't have enough time to validate that we are going to sleep the second time around the event loop. My assumption was that this was a level triggered interrupt. If it is indeed edge triggered, we can ack the interrupt from the event loop. Will look later today

Reply:
> I didn't have enough time yesterday to validate that we are going to sleep the second time around the event loop. My assumption was that the mailbox notification was a level triggered interrupt. If it is indeed edge triggered, we can ack the interrupt from the event loop. Will look later today

It is level triggered, but it's sticky I believe. https://chipsalliance.github.io/caliptra-rtl/main/internal-regs/?p=clp.soc_ifc_reg.intr_block_rf.error_internal_intr_r

Reply:
The status bit needs to be sticky, but the question is whether the actual signal itself is...

Reply:
Looks like this is effectively edge-triggered. The interrupt register is set with a pulse here:

https://github.com/chipsalliance/caliptra-rtl/blob/bbc9a329abb109abee5befc39419a5336672d036/src/soc_ifc/rtl/soc_ifc_top.sv#L664

And that register appears to be used to build the soc_ifc_notif_intr wire going to the cpu.

Reply:
Finally got my FPGA setup working again so I could validate it actually went to sleep. Turns out the interrupt was still high from FIRMWARE_LOAD, so I needed to handle that to allow it to sleep before the first runtime command.